From 0dfe6c00d363d037dd484250fd015af719d579ce Mon Sep 17 00:00:00 2001
From: Erico Nunes <nunes.erico@gmail.com>
Date: Wed, 22 Jan 2020 00:37:22 +0100
Subject: [PATCH 062/102] lima/ppir: fix ssa undef emit

The ssa doesn't need to be manually added to block->comp->reg_list.
Doing so actually causes other registers to be marked as undef=true
later.

This patch alone fixes a few deqp tests that have undefs.

Signed-off-by: Erico Nunes <nunes.erico@gmail.com>
Reviewed-by: Vasily Khoruzhick <anarsoul@gmail.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/merge_requests/3502>
---
 src/gallium/drivers/lima/ir/pp/nir.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/gallium/drivers/lima/ir/pp/nir.c b/src/gallium/drivers/lima/ir/pp/nir.c
index 0c91c09831b..9798d16dd1e 100644
--- a/src/gallium/drivers/lima/ir/pp/nir.c
+++ b/src/gallium/drivers/lima/ir/pp/nir.c
@@ -440,9 +440,6 @@ static ppir_node *ppir_emit_ssa_undef(ppir_block *block, nir_instr *ni)
 
    ppir_dest *dest = &alu->dest;
    dest->ssa.undef = true;
-   ppir_reg *ssa = &dest->ssa;
-
-   list_add(&ssa->list, &block->comp->reg_list);
 
    return node;
 }
-- 
2.24.1

